parser.y

```diff
@@ -2434,7 +2448,64 @@ IndexColNameList:
 		$$ = append($1.([]*ast.IndexColName), $3.(*ast.IndexColName))
 	}
 
+IndexLockAndAlgorithmOpt:
+	{
+		$$ = &ast.IndexLockAndAlgorithm{
```

Comment:
we could use `[2]interface{}` to hold two kind of value.
in my opinion, it's not necessary to export a new struct

Reply:
Yes. And I will reuse it in #459 and #460.